mycore-mods/src/main/resources/xsl/mods-solr.xsl

```diff
@@ -95,9 +116,16 @@
     <xsl:for-each
       select=".//mods:originInfo[not(@eventType) or @eventType='publication']/mods:publisher">
       <field name="mods.publisher">
-        <xsl:for-each select="mods:displayForm | mods:namePart | text()">
-          <xsl:value-of select="concat(' ',.)" />
-        </xsl:for-each>
+        <xsl:choose>
+          <xsl:when test="mods:displayForm">
+            <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(mods:displayForm))" />
```

Comment:
same here: no 'concat' needed

probably mods:name/mods:publisher deserves a custom template that returns a value for an index field

Reply:
I took a look in my index and I think there is no need for this. We use a standard solr textfield here with tokenizer.